src/jquery.fancytree.fixed.js

```diff
@@ -21,62 +21,449 @@
+		});
+		
+		return res;
+	},
+	treeLoad: function(ctx) {
+		var _this = this,
+			res = this._superApply(arguments);
+		
+		res.done(function() {
+			_this.ext.fixed._adjustLayout();
+		});
+		return res;
 	},
 	/* Called by nodeRender to sync node order with tag order.*/
-//    nodeFixOrder: function(ctx) {
-//    },
+//	nodeFixOrder: function(ctx) {
+//	},
+	
+	nodeLoadChildren: function(ctx, source) {
+		return this._superApply(arguments);
+	},
 	nodeRemoveChildMarkup: function(ctx) {
 		return this._superApply(arguments);
 	},
 	nodeRemoveMarkup: function(ctx) {
 		return this._superApply(arguments);
 	},
+	nodeSetActive: function(ctx, flag) {
+		var node = ctx.node,
+			$rightNode = $(node.tr),
+			$leftNode = $rightNode.data("fancytree-node-counterpart");
+		$leftNode.toggleClass("fancytree-active", flag);
+		var res = this._superApply(arguments);
+		return res;
+	},
 	nodeRender: function(ctx, force, deep, collapsed, _recursive) {
-		return this._superApply(arguments);
+		var res = this._superApply(arguments),
+			node = ctx.node,
+			$rightTr = $(node.tr),
+			$leftTr = $(node.tr).data("fancytree-node-counterpart");
+		if ($leftTr) {
+			$leftTr.attr("class", $rightTr.attr("class"));
+		}
+		return res;
 	},
 	nodeRenderTitle: function(ctx, title) {
 		return this._superApply(arguments);
 	},
 	nodeRenderStatus: function(ctx) {
-		return this._superApply(arguments);
+		var res = this._superApply(arguments),
+			node = ctx.node,
+			$rightTr = $(node.tr).data("fancytree-node-counterpart");
+		if ($rightTr) {
+			$rightTr.toggleClass("fancytree-selected", node.selected);
+			$rightTr.toggleClass("fancytree-partsel", node.partsel);
+		}
+		
+		return res;
 	},
 	nodeSetExpanded: function(ctx, flag, opts) {
-		return this._superApply(arguments);
+		var _this = this,
+			node = ctx.node,
+			fixCols = this.options.fixed.fixCols,
+			$leftTr = $(node.tr).data("fancytree-node-counterpart"),
+			res = this._superApply(arguments);
+		$leftTr.toggleClass("fancytree-expanded", flag);
+		res.done(function() {
+			node.visit(function(child) {
+				var $tr = $(child.tr),
+					$clone = $tr.data("fancytree-node-counterpart");
+				$clone.toggleClass("hidden", !flag).find("td, th").toggleClass("hidden", !flag);
+				_this.ext.fixed._adjustRowLayout($tr, $clone);
+				if (!child.expanded) {
+					return "skip";
+				}
+			});
+			
+			$leftTr.find("td:not(.hidden), th:not(.hidden)").each(function(idx) {
+				_this.ext.fixed._adjustColumnLayout($(this), null);
+			});
+			_this.ext.fixed._adjustWrapperLayout();
+		});
+		return res;
 	},
 	nodeSetStatus: function(ctx, status, message, details) {
+		console.log("set status", arguments);
 		return this._superApply(arguments);
 	},
 	treeClear: function(ctx) {
+		var tree = ctx.tree,
+			$table = tree.widget.element,
+			$wrapper = this.$fixedWrapper;
+		$table.find("tr, td, th, thead").removeClass("hidden").css({
+			"min-width": "auto",
+			"height": "auto"
+		});
+		$wrapper.append($table);
```

Comment:
I think we should discard the markup on treeDestroy, but not treeClear.
Only remove all rows here?